src/main/java/org/logstash/dissect/ValueRef.java

```diff
@@ -1,35 +1,21 @@
 package org.logstash.dissect;
 
-import java.util.Arrays;
+final class ValueRef {
+    private final int position;
+    private final int length;
 
-public class ValueRef {
-    private int position = 0;
-    private int length = 0;
-
-    public void set(int position, int length) {
+    ValueRef(int position, int length) {
         this.position = position;
         this.length = length;
     }
 
-    public void setPosition(int position) {
-        this.position = position;
-    }
-
-    public void setLength(int length) {
-        this.length = length;
-    }
-
-    public String string(byte[] source) {
+    String extract(byte[] source) {
         return new String(source, position, length);
```

Comment:
You're relying on the default platform encoding here. If you know this needs to be a UTF-8 string, then you should add `StandardCharsets.UTF_8` as 4th parameter